engines/sci/engine/message.cpp

```diff
@@ -444,6 +444,15 @@ bool MessageState::stringStage(Common::String &outstr, const Common::String &inS
 		// SCI32 seems to support having digits in stage directions
 		if (((inStr[i] >= 'a') && (inStr[i] <= 'z')) || ((inStr[i] >= '0') && (inStr[i] <= '9') && (getSciVersion() < SCI_VERSION_2)))
 			return false;
+
+		// If it contains Hebrew letters, it's not a stage direction
+		// besides that point, keep the English logic
+		if (g_sci->getLanguage() == Common::HE_ISR) {
+			if (((inStr[i] >= 'a') && (inStr[i] <= 'z')) ||
+				(((byte)inStr[i]) >= 128 && ((byte)inStr[i] <= 255)) ||
+				((inStr[i] >= '0') && (inStr[i] <= '9') && (getSciVersion() < SCI_VERSION_2)))
```

Comment:
You don't need to repeat the same checks as the ones in the check before this one.
This can be simplified as:

    if (g_sci->getLanguage() == Common::HE_ISR && getSciVersion() < SCI_VERSION_2 && (byte)inStr[i] >= 128 && (byte)inStr[i] <= 255)
         return false;

Reply:
Makes sense.
But I'm not sure regarding the `getSciVersion() < SCI_VERSION_2`.

If we examine the original English check (with added `newline` ):
```
		// If we find a lowercase character or a digit, it's not a stage direction
		// SCI32 seems to support having digits in stage directions
		if (((inStr[i] >= 'a') && (inStr[i] <= 'z')) || 
		    ((inStr[i] >= '0') && (inStr[i] <= '9') && (getSciVersion() < SCI_VERSION_2)))
			return false;
```
it's of the kind `if A || B`, and the version detection is only needed for `B`, while the letter comparison is done regardless of the version. (as also explained in the comment)

Therefore, I think my PR should be further simiplified to:
```
		if (g_sci->getLanguage() == Common::HE_ISR && (byte)inStr[i] >= 128 && (byte)inStr[i] <= 255)
			return false;
```

Reply:
Indeed, it can be simplified to just that. Please do so, and we'll get this merged :)

Reply:
Great.
I've pushed it.

Reply:
Thanks, merged